Source/WebCore/html/DOMURL.cpp

```diff
@@ -50,15 +50,15 @@ ExceptionOr<Ref<DOMURL>> DOMURL::create(const String& url, const URL& base)
     ASSERT(base.isValid() || base.isNull());
     URL completeURL { base, url };
     if (!completeURL.isValid())
-        return Exception { TypeError };
+        return Exception { TypeError, makeString("URL Constructor: ", url, " is not a valid URL.") };
```

Comment:
Although our internal API uses `isValid()`, that's not an accurate reflection of what the web standard calls valid. So the web developer facing message should be something like "can't be parsed" here and below.

And perhaps below we should also log the base? As in, "{url} cannot be parsed against {base}."?

Reply:
I'm happy to have more meaningful messages. The more explicit for the Webdev, the better.

Reply:
so there are two different `DOMURL::create`

## `completeURL`

https://github.com/WebKit/WebKit/blob/73de3248842cc21f0547553fda5694324c4ce219/Source/WebCore/html/DOMURL.cpp#L48-L55

This one kicks in for: 

* `new URL("/")`

## `baseURL`

https://github.com/WebKit/WebKit/blob/73de3248842cc21f0547553fda5694324c4ce219/Source/WebCore/html/DOMURL.cpp#L57-L63

this is calling eventually the other `DOMURL::create` if the `baseURL` is not `null` and `valid`.

* `new URL("/", "/")`
* `new URL("/", "")`



Also `canParse` seems like dead code, probably to fix in another PR.

https://github.com/WebKit/WebKit/blob/73de3248842cc21f0547553fda5694324c4ce219/Source/WebCore/html/DOMURL.cpp#L67-L74
https://searchfox.org/wubkat/search?q=symbol:_ZN7WebCore6DOMURL8canParseERKN3WTF6StringES4_&redirect=false

Reply:
canParse was added by @annevk so not dead code.

Reply:
<img width="798" alt="console rendering of error messages" src="https://github.com/WebKit/WebKit/assets/505230/fd4365bf-a830-4a2e-bc96-7ce48306f892">

This is the result of my modifications, I will adjust the tests.